plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java

```diff
@@ -280,13 +279,13 @@ private void triggerModelRedeploy(ModelAutoRedeployArrangement modelAutoRedeploy
         String[] nodeIds = null;
         if (deployToAllNodes || !allowCustomDeploymentPlan) {
             nodeIds = new String[0];
-        } else if (planningWorkerNodes != null && planningWorkerNodes.size() > 0) {
+        } else if (planningWorkerNodes != null && !planningWorkerNodes.isEmpty()) {
             // allow custom deploy plan and not deploy to all case, we need to check if the added nodes in planning worker nodes.
             List<String> needRedeployPlanningWorkerNodes = Arrays
                 .stream(planningWorkerNodes.toArray(new String[0]))
                 .filter(addedNodes::contains)
-                .collect(Collectors.toList());
-            nodeIds = needRedeployPlanningWorkerNodes.size() > 0 ? planningWorkerNodes.toArray(new String[0]) : null;
+                .toList();
```

Comment:
This works on JDK21 but since this will be backported you may want to keep the old version

Reply:
good call out.